test/src/com/xilinx/rapidwright/eco/TestECOTools.java

```diff
@@ -0,0 +1,438 @@
+
+        String instName = ehciToDelete.getFullHierarchicalInstName();
+
+        // Logical leaf cell not present
+        Assertions.assertNull(netlist.getHierCellInstFromName(instName));
+        // Physical cell not present
+        Assertions.assertNull(design.getCell(instName));
+
+        DesignTools.batchRemoveSitePins(deferredRemovals, true);
+
+        if (FileTools.isVivadoOnPath()) {
+            ReportRouteStatusResult rrs = VivadoTools.reportRouteStatus(design);
+            Assertions.assertEquals(8, rrs.netsWithRoutingErrors);
+            Assertions.assertEquals(8, rrs.netsWithNoDriver);
+            Assertions.assertEquals(8, rrs.netsWithSomeUnroutedPins);
+        }
+    }
+
+    @Test
+    public void testCreateCell() {
+        Design design = RapidWrightDCP.loadDCP("picoblaze_ooc_X10Y235.dcp");
+        EDIFNetlist netlist = design.getNetlist();
+
+        DesignTools.updatePinsIsRouted(design);
+
+        EDIFCell reference = netlist.getCell("kcpsm6");
+        List<String> instNames = Arrays.asList("processor2", "processor3");
+        ECOTools.createCell(design, reference, instNames);
+
+        List<EDIFHierCellInst> goldenLeaves = netlist.getAllLeafDescendants("processor");
+
+        for (String instName : instNames) {
+            // Logical hierarchical cell is present
+            EDIFHierCellInst ehci = netlist.getHierCellInstFromName(instName);
+            Assertions.assertNotNull(ehci);
+
+            // Physical leaf cells are present and unplaced
+            List<EDIFHierCellInst> leaves = netlist.getAllLeafDescendants(ehci);
+            Assertions.assertEquals(goldenLeaves.size(), leaves.size());
+            for (EDIFHierCellInst leaf : leaves) {
+                String cellName = leaf.getCellName();
+                if (cellName.equals("VCC") || cellName.equals("GND")) {
+                    continue;
+                }
+                String leafName = leaf.getFullHierarchicalInstName();
+                Cell leafCell = design.getCell(leafName);
+                Assertions.assertNotNull(leafCell);
+                Assertions.assertFalse(leafCell.isPlaced());
+            }
+        }
+
+        if (FileTools.isVivadoOnPath()) {
+            ReportRouteStatusResult rrs = VivadoTools.reportRouteStatus(design);
+            Assertions.assertEquals(1135, rrs.logicalNets);
+            Assertions.assertEquals(728, rrs.netsWithNoPlacedPins);
+            Assertions.assertEquals(2, rrs.netsWithRoutingErrors);
+            Assertions.assertEquals(2, rrs.netsWithSomeUnplacedPins);
+        }
+    }
+
+    @Test
+    public void testCreateNet() {
+        Design design = RapidWrightDCP.loadDCP("picoblaze_ooc_X10Y235.dcp");
+        EDIFNetlist netlist = design.getNetlist();
+
+        DesignTools.updatePinsIsRouted(design);
+
+        List<String> netNames = Arrays.asList("processor/foo", "your_program/bar");
+        ECOTools.createNet(design, netNames);
```

Comment:
Should we be checking the return value?

Reply:
Same as for `createCell` -- can return a list of `EDIFHierNet`s if that makes sense?